src/validation_workflow/docker/validation_docker.py

```diff
@@ -56,121 +59,131 @@ def start_cluster(self) -> bool:
-        result = subprocess.run(docker_compose_up, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
-        return ('returncode=0' in (str(result)), self.target_yml_file)
-
-    def inplace_change(self, filename: str, old_string: str, new_string: str) -> None:
-
-        with open(filename) as f:
-            s = f.read()
-            if old_string not in s:
-                print('"{old_string}" not found in {filename}.'.format(**locals()))
-                return
+        for url, name in self.test_readiness_urls.items():
+            try:
+                status_code, response_text = ApiTest(url).api_get()
+                if status_code != 200:
+                    logging.error(f'Error connecting to {name} ({url}): status code {status_code}')
+                    return False
+            except (requests.exceptions.ConnectionError, requests.exceptions.ConnectTimeout) as e:
+                logging.error(f'Error connecting to {name} ({url}): {e}')
+                return False
+        return True
 
-        with open(filename, 'w') as f:
-            print('Changing "{old_string}" to "{new_string}" in {filename}'.format(**locals()))
-            s = s.replace(old_string, new_string)
-            f.write(s)
+    def check_cluster_readiness(self) -> bool:
+        max_retry = 20
+        retry_count = 0
+        while retry_count < max_retry:
+            logging.info(f'sleeping 10sec for retry {retry_count + 1}/{max_retry}')
+            time.sleep(10)
+            if self.check_http_request():
+                logging.info('\n\ncluster is now ready for API test\n\n')
+                return True
+            retry_count += 1
+        logging.error(f"Maximum number of retries ({max_retry}) reached. Cluster is not ready for API test.")
+        return False
+
+    def get_artifact_image_name(self, artifact: str, using_staging_artifact_only: str) -> Any:
+        self.image_names = {
+            'dockerhub': {
+                'opensearch': {
+                    'staging': 'opensearchstaging/opensearch',
+                    'production': 'opensearchproject/opensearch'
+                },
+                'opensearch_dashboards': {
+                    'staging': 'opensearchstaging/opensearch-dashboards',
+                    'production': 'opensearchproject/opensearch-dashboards'
+                }
+            },
+            'ecr': {
+                'opensearch': {
+                    'staging': 'public.ecr.aws/opensearchstaging/opensearch',
+                    'production': 'public.ecr.aws/opensearchproject/opensearch'
+                },
+                'opensearch_dashboards': {
+                    'staging': 'public.ecr.aws/opensearchstaging/opensearch-dashboards',
+                    'production': 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+                }
+            }
+        }
+        return self.image_names[self.args.docker_source][artifact][using_staging_artifact_only]
```

Comment:
You should make this OO. Create a class called `DockerImage` that takes source, artifact and staging as input and generates the name as a method and returns `public.ecr.aws/opensearchstaging/opensearch-dashboards` and such as a property. Each one of the parts of the result has meaning. For example a method called `host` would return `null` if the registry is dockerhub, public.ecr.aws if it's ecr, etc.